tests/PHPStan/Rules/Operators/InvalidBinaryOperationRuleTest.php

```diff
@@ -659,6 +665,14 @@ public function testBenevolentUnion(): void
 				'Binary operation "/" between (array<string, string>|bool|int|object|resource) and BinaryOpBenevolentUnion\\Foo results in an error.',
 				105,
 			],
+			[
+				'Binary operation "/" between (array<string, string>|bool|int|object|resource) and (array<string, string>|bool|int|object|resource) might result in an error.',
```

Comment:
This one. It might be `int / int` so this shouldn't be reported.

Reply:
Oh I get it, but `int` can still include zero so this should be reported.

But: We can still create an artifical example for example of `(float|int)` that shouldn't be reported because we don't report `X / float`.